app/src/androidTest/java/com/example/bigowlapp/activity/SignUpPageActivityTest.java

```diff
@@ -0,0 +1,202 @@
+package com.example.bigowlapp.activity;
+
+import android.os.SystemClock;
+import android.view.View;
+import android.view.ViewParent;
+import android.widget.FrameLayout;
+
+import androidx.core.widget.NestedScrollView;
+import androidx.test.espresso.Espresso;
+import androidx.test.espresso.PerformException;
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.espresso.util.HumanReadables;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+import androidx.test.rule.ActivityTestRule;
+
+import com.example.bigowlapp.R;
+import com.example.bigowlapp.viewModel.SignUpViewModel;
+
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.Random;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.scrollTo;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.hasErrorText;
+import static androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom;
+import static androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.isRoot;
+import static androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class SignUpPageActivityTest {
```

Comment:
The current coverage is about 63% for this class. Looking at the code coverage of the class, I think you can still add more tests that require little effort which can increase coverage even further.

![image](https://user-images.githubusercontent.com/23643192/112911358-4379a480-90c3-11eb-8f26-0259554e7238.png)

The one that I'm talking about is the error handling. It should be simple and it would cover lots of lines.

You would have code like the one in ScheduleFormFragmentTest: 
![image](https://user-images.githubusercontent.com/23643192/112910975-74a5a500-90c2-11eb-82a5-13bba9d734b8.png)

I would also add tests that would simply click the tvSignIn button.

Another thing, if it's possible, then I think mocking the viewModel would be beneficial.

Reply:
I did an extra test for tvSignIn and I made a mocking now.